mobile-browser-based-version/src/core/dataset/data_loader/data_loader.ts

```diff
@@ -0,0 +1,10 @@
+import { Dataset } from '../dataset_builder'
+
+export type Source = string | File
```

Comment:
Might be a good idea to create another type for string, string at the end will hold a url right?

so maybe we could do

export type StringURL = string
export type Source = StringURL | File

and then do the changes accordingly, e.g in `image_loader.ts`
`if (typeof image === 'string')` -> `if (typeof image === StringURL')`
I also think the latter is more clear when reading the code.

Reply:
@tharvik , at the end with Hugo we agreed that a `type Path = string` was a good idea, however it seems we cannot use it for type checking, i.e. `source instanceof Path` nor `source typeof === Path` work, where `source` if of type `Source`, which is defined as `type Source = string | File`.

It [seems](https://stackoverflow.com/questions/62769751/can-i-determine-if-a-value-matches-a-type-alias) that types are erased at runtime, so there is no way to check it. However this seems too bad since it would make the code easier to read. Do you know if there is another way to check types here?